sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -225,6 +219,151 @@ case class GpuParquetPartitionReaderFactory(
       version >= "3.0.0" && lookupFileMeta(SPARK_LEGACY_DATETIME) == null
     }.getOrElse(isCorrectedModeConfig)
   }
+}
+
+private case class MultiFilePartitionInfo(filePath: Path, blockMeta: BlockMetaData,
+    partValues: InternalRow, schema: MessageType)
+
+/**
+ * Similar to GpuParquetPartitionReaderFactory but extended for reading multiple files
+ * in an iteration. This will allow us to read multiple small files and combine them
+ * on the CPU side before sending them down to the GPU.
+ */
+case class GpuParquetMultiFilePartitionReaderFactory(
+    @transient sqlConf: SQLConf,
+    broadcastedConf: Broadcast[SerializableConfiguration],
+    dataSchema: StructType,
+    readDataSchema: StructType,
+    partitionSchema: StructType,
+    filters: Array[Filter],
+    @transient rapidsConf: RapidsConf,
+    metrics: Map[String, SQLMetric]) extends PartitionReaderFactory with Arm with Logging {
+  private val isCaseSensitive = sqlConf.caseSensitiveAnalysis
+  private val enableParquetFilterPushDown: Boolean = sqlConf.parquetFilterPushDown
+  private val pushDownDate = sqlConf.parquetFilterPushDownDate
+  private val pushDownTimestamp = sqlConf.parquetFilterPushDownTimestamp
+  private val pushDownDecimal = sqlConf.parquetFilterPushDownDecimal
+  private val pushDownStringStartWith = sqlConf.parquetFilterPushDownStringStartWith
+  private val pushDownInFilterThreshold = sqlConf.parquetFilterPushDownInFilterThreshold
+  private val debugDumpPrefix = rapidsConf.parquetDebugDumpPrefix
+  private val maxReadBatchSizeRows = rapidsConf.maxReadBatchSizeRows
+  private val maxReadBatchSizeBytes = rapidsConf.maxReadBatchSizeBytes
```

Comment:
Nit: There's a lot of code related to predicate pushdown that's replicated between the multi- and single file factory forms.  Would be nice to commonize this if clean to do so, maybe the base class sets up everything to do with the filters and has a protected method to filter the row groups that the multi- and single- forms can both leverage.